keras_cv/layers/object_detection/object_detection_prediction_decoder.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+from keras_cv import bounding_box
+from keras_cv import layers as cv_layers
+
+
+class ObjectDetectionPredictionDecoder(tf.keras.layers.Layer):
+    """A Keras layer that decodes predictions of an object detection model.
+
+    By default, ObjectDetectionPredictionDecoder uses a
+    `keras_cv.layers.NonMaxSuppression` layer to perform box pruning.
+
+    Attributes:
+      classes: Number of classes in the dataset.
+      bounding_box_format: The format of bounding boxes of input dataset. Refer
+        [to the keras.io docs](https://keras.io/api/keras_cv/bounding_box/formats/)
+        for more details on supported bounding box formats.
+      anchor_generator:
+      suppression_layer:
+      box_variance:
+    """
+
+    def __init__(
+        self,
+        bounding_box_format,
+        anchor_generator,
+        classes=None,
+        suppression_layer=None,
+        box_variance=(0.1, 0.1, 0.2, 0.2),
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        if not suppression_layer and not classes:
+            raise ValueError(
+                "ObjectDetectionPredictionDecoder() requires either `suppression_layer` "
+                f"or `classes`.  Received `suppression_layer={suppression_layer} and "
+                f"classes={classes}`"
+            )
+        self.bounding_box_format = bounding_box_format
+        self.suppression_layer = suppression_layer or cv_layers.NonMaxSuppression(
+            classes=classes,
+            bounding_box_format=bounding_box_format,
+            confidence_threshold=0.05,
+            iou_threshold=0.5,
+            max_detections=100,
+            max_detections_per_class=100,
+        )
+        if self.suppression_layer.bounding_box_format != self.bounding_box_format:
```

Comment:
It seems that user might not be able to pass random layers as suppression_layer. We should document that.

Reply:
Documented